ipaserver/plugins/idrange.py

```diff
@@ -482,6 +496,15 @@ def pre_callback(self, ldap, dn, entry_attrs, attrs_list, *keys, **options):
                             'or ipa-ad-trust-posix when SID of the trusted '
                             'domain is not specified.'))
 
+            # auto private group is possible only for ad trusts
+            if entry_attrs['iparangetype'] == 'ipa-local':
```

Comment:
For future proofing rather than looking for the odd-one out would this be better:
if entry_attrs['iparangetype'] not in ('ipa-ad-trust', 'ipa-ad-trust-posix'):

Is it safe to pull this directly and not use entry_attrs.get('iparangetype')?

Reply:
> For future proofing rather than looking for the odd-one out would this be better:
> if entry_attrs['iparangetype'] not in ('ipa-ad-trust', 'ipa-ad-trust-posix'):

Good suggestion, updated.

> Is it safe to pull this directly and not use entry_attrs.get('iparangetype')?
Yes it's safe thanks to these lines just a few lines above:
```
            # Default to ipa-local if no type set
            if 'iparangetype' not in entry_attrs:
                entry_attrs['iparangetype'] = 'ipa-local'
```